vortex-io/src/file/object_store.rs

```diff
@@ -129,10 +130,18 @@ impl ReadSource for ObjectStoreIoSource {
                             // The read_exact_at call will either fill the entire buffer or return an error,
                             // ensuring no uninitialized memory is exposed.
                             unsafe { buffer.set_len(len) };
+
                             handle
                                 .spawn_blocking(move || {
-                                    file.read_exact_at(&mut buffer, range.start)?;
-                                    Ok::<_, io::Error>(buffer)
+                                    #[cfg(unix)] {
+                                        file.read_exact_at(&mut buffer, range.start)?;
+                                        Ok::<_, io::Error>(buffer)
+                                    }
+                                    #[cfg(not(unix))] {
+                                        file.seek(range.start)?;
+                                        file.read_exact(&mut buffer)?;
```

Comment:
state of these confuses me deeply in rust standard library. I thought there's an equivalent read_exact_at but no you have to map and unmap the file which would be bad